File: Frontend/src/services/irrigationService.ts
import api from './api';
import type { IrrigationControlRequest, IrrigationResponse } from '../types';

export const irrigationService = {
  start: (farmerId: number, data?: IrrigationControlRequest) =>
    api.post<IrrigationResponse>(`/irrigation/farmers/${farmerId}/start`, data ?? {}),
  stop: (farmerId: number, data?: IrrigationControlRequest) =>
    api.post<IrrigationResponse>(`/irrigation/farmers/${farmerId}/stop`, data ?? {}),
  history: (farmerId: number) =>
    api.get<IrrigationResponse[]>(`/irrigation/farmers/${farmerId}/history`),
};

File: Backend/src/main/java/com/smartagriculture/service/IrrigationService.java
package com.smartagriculture.service;

import com.smartagriculture.dto.irrigation.IrrigationControlRequest;
import com.smartagriculture.dto.irrigation.IrrigationResponse;
import java.util.List;

public interface IrrigationService {

    IrrigationResponse start(Long farmerId, IrrigationControlRequest request);

    IrrigationResponse stop(Long farmerId, IrrigationControlRequest request);

    List<IrrigationResponse> history(Long farmerId);
}
